source/common/network/utility.cc

```diff
@@ -156,6 +156,16 @@ void Utility::throwWithMalformedIp(const std::string& ip_address) {
   throw EnvoyException(fmt::format("malformed IP address: {}", ip_address));
 }
 
+Address::InstanceConstSharedPtr
+Utility::mapIPv4toIPv6(const Address::InstanceConstSharedPtr& address) {
+  if (address->ip()->version() == Address::IpVersion::v4) {
+    return std::make_shared<Address::Ipv6Instance>("::ffff:" + address->ip()->addressAsString(),
```

Comment:
perf nit: Is there a constructor version we can use where we can potentially just operate on the raw bytes? It seems like you can just take a uint128_t and mask in certain bytes?

Reply:
will do